dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_disjointBlockSet_addBlock(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		disjointBlockSet disjointBlockSet
+		block            *types.Block
+		err              error
+	}{
+		{
+			name: "add block beyond capacity",
+			block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			},
+			err: errors.New("cannot add block; set is at capacity"),
+		},
+		{
+			name: "add block",
+			disjointBlockSet: disjointBlockSet{
+				limit:            1,
+				blocks:           make(map[common.Hash]*pendingBlock),
+				timeNow:          time.Now,
+				parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
+			},
+			block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			},
+			err: nil,
+		},
+	}
+	for _, tt := range tests { //nolint:govet
+		tt := tt //nolint:govet
```

Comment:
nit change your test case to use a `*disjointBlockSet` instead of `disjointBlockSet` and remove those `//nolint:govet`. That way the mutex lock isn't 'copied' and the linter won't complain.

Reply:
Updated.